include/albatross/src/evaluation/cross_validation_utils.hpp

```diff
@@ -93,15 +93,25 @@ inline MarginalDistribution concatenate_marginal_predictions(
   Eigen::VectorXd variance(n);
   Eigen::Index number_filled = 0;
   // Put all the predicted means back in order.
+  bool has_covariance = false;
   for (const auto &pair : indexer) {
     assert(preds.at(pair.first).size() == pair.second.size());
     set_subset(preds.at(pair.first).mean, pair.second, &mean);
-    set_subset(preds.at(pair.first).covariance.diagonal(), pair.second,
-               &variance);
+    if (preds.at(pair.first).has_covariance()) {
```

Comment:
Do we want to do this or just add `1e6` as variances where we currently have none?

Reply:
In this method we're taking a pair of `MarginalDistribution` and concatenating them, so if the both don't have a defined covariance then we want to preserve that in the concatenation.  Somewhere on my list of want to dos is to remove the optional behavior for convariances in favor of a third distribution type, something like:
```
using MeanOnlyDistribution = Distribution<Empty>;
using MarginalDistribution = Distribution<DiagonalMatrixXd>;
using JointDistribution = Distribution<Eigen::MatrixXd>;
```
or something along those lines, but that's out of scope here.